game/src/layer/traffic.rs

```diff
@@ -596,3 +597,90 @@ impl Jam {
         polygons
     }
 }
+
+pub struct DelayPerAgent {
+    time: Time,
+    unzoomed: Drawable,
+    panel: Panel,
+}
+
+impl Layer for DelayPerAgent {
+    fn name(&self) -> Option<&'static str> {
+        Some("delay per agent")
+    }
+    fn event(
+        &mut self,
+        ctx: &mut EventCtx,
+        app: &mut App,
+        minimap: &Panel,
+    ) -> Option<LayerOutcome> {
+        if app.primary.sim.time() != self.time {
+            *self = DelayPerAgent::new(ctx, app);
+        }
+
+        self.panel.align_above(ctx, minimap);
+        match self.panel.event(ctx) {
+            Outcome::Clicked(x) => match x.as_ref() {
+                "close" => {
+                    return Some(LayerOutcome::Close);
+                }
+                _ => unreachable!(),
+            },
+            _ => {}
+        }
+        None
+    }
+    fn draw(&self, g: &mut GfxCtx, app: &App) {
+        self.panel.draw(g);
+        if g.canvas.cam_zoom < app.opts.min_zoom_for_detail {
+            g.redraw(&self.unzoomed);
+        }
+    }
+    fn draw_minimap(&self, g: &mut GfxCtx) {
+        g.redraw(&self.unzoomed);
+    }
+}
+
+impl DelayPerAgent {
+    pub fn new(ctx: &mut EventCtx, app: &App) -> DelayPerAgent {
+        let mut delays = app.primary.sim.all_waiting_people();
+        let mut unzoomed = GeomBatch::new();
+        unzoomed.push(
```

Comment:
Kind of a bug here: if you have the layer open and click an agent, two different batches apply this fade, and the map becomes very dark. Not really an issue before, because there aren't obvious ways how to apply two layer-ish things at the same time -- but also hard to verify that.